multi-agent/lib/mas/resources/builtin_aware_service.py

```
@@ -0,0 +1,532 @@
+        return result
+
+    # ── Re-implemented methods (self-calls must route through decorator) ─
+    #
+    # These methods exist on CoreResourceService but call overridden methods
+    # internally (get_visible, resolve_resource, _build_configs_from_rids).
+    # Delegating to self._inner would bypass the decorator's overrides, so
+    # they are re-implemented here with self.X() calls that route through
+    # the decorator's versions.  Internal state is accessed via the inner.
+
+    def resolve(
+        self, rid: str, caller: CallerScope = CallerScope(),
```

Ruff (0.16.1)

[warning] Do not perform function call `CallerScope` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)